packages/docker-node/src/docker-manager.ts

```diff
@@ -0,0 +1,162 @@
+import Docker, { ImageBuildContext } from 'dockerode';
+import fsExtra from 'fs-extra';
+import { IncomingMessage } from 'http';
+import {
+  DockerBadGatewayError,
+  DockerHubConnectionError,
+  DockerNotInstalledError,
+  DockerNotRunningError,
+  DockerServerError,
+  ExecutableNotFoundError,
+  ImageDoesntExistError,
+} from './errors';
+import { Image } from './types';
+
+const DOCKER_SOCKET_PATH = '/var/run/docker.sock';
+
+export class DockerManager {
+  public static async create(): Promise<DockerManager> {
+    if (!(await DockerManager.isInstalled())) {
+      throw new DockerNotInstalledError();
+    }
+
+    if (!(await fsExtra.pathExists(DOCKER_SOCKET_PATH))) {
+      throw new DockerNotRunningError();
+    }
+
+    return new DockerManager(Docker);
+  }
+
+  public static async isInstalled(): Promise<boolean> {
+    const { exec } = await import('child_process');
+    return new Promise((resolve) => {
+      exec('which docker', (error?) => resolve(!error));
+    });
+  }
+
+  public static imageToRepoTag(image: Image): string {
+    return `${image.repository}:${image.tag}`;
+  }
+
+  readonly _docker: Docker;
+
+  // The constructor is private, see [[DockerManager.create]].
+  private constructor(DockerImpl: typeof Docker) {
+    this._docker = new DockerImpl({ socketPath: DOCKER_SOCKET_PATH });
+  }
+
+  public async isRunning(): Promise<boolean> {
+    try {
+      const result = await this._withCommonErrors(this._docker.ping());
+      return result === 'OK';
+    } catch (error) {
+      if (error instanceof DockerNotRunningError) {
+        return false;
+      }
+
+      if (error instanceof DockerBadGatewayError) {
+        return false;
+      }
+
+      throw error;
+    }
+  }
+
+  public async imageExists(image: Image): Promise<boolean> {
+    const { got } = await import('got-cjs');
```

Comment:
Is `got-cjs` a special http client?

I ask this because another HTTP client `cross-fetch` was introduced in PR https://github.com/ckb-js/kuai/pull/37 and it has a native node module `fetch`.

Is it possible to use native `fetch` instead of the `got-cjs` package?